### PR TITLE
Add some examples for graph introspection

### PR DESCRIPTION
**Pull request by jamoralp**

### CODE REVIEW

File: rclc/graph/main.c [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  RCCHECK(rcl_wait_set_init(&wait_set, 0, 1, 0, 0, 0, 0, &context, allocator));
+
+  while (true)
+  {
+    RCSOFTCHECK(rcl_wait_set_clear(&wait_set));
+
+    size_t index;
+
+    const rcl_guard_condition_t * graph_guard_condition =
+      rcl_node_get_graph_guard_condition(&node);
```
Comment (pablogs9) [inline rclc/graph/main.c:34]: Maybe we can retrieve the GC once

File: rclc/publisher_count/main.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    &support,
+    RCL_MS_TO_NS(timer_timeout),
+    timer_callback));
+
+  // create executor
+  rclc_executor_t executor = rclc_executor_get_zero_initialized_executor();
+  RCCHECK(rclc_executor_init(&executor, &support.context, 1, &allocator));
+
+  unsigned int rcl_wait_timeout = 1000;
+  RCCHECK(rclc_executor_set_timeout(&executor, RCL_MS_TO_NS(rcl_wait_timeout)));
```
Comment (pablogs9) [inline rclc/publisher_count/main.c:63]: Remove this here and in the others

**Comment by pablogs9** [comment]
What about putting this 3 examples in a subfolder inside `rclc`, just like the configured examples folder?

**Comment by jamoralp** [comment]
> What about putting this 3 examples in a subfolder inside `rclc`, just like the configured examples folder?

Sounds totally right to me

**Comment by pablogs9** [comment]
Ok, I'm merging when the examples are in their folder

**Review by pablogs9 — APPROVED** [review]

**Pull request merged**